src/utils/ThemePreference.tsx

```diff
@@ -1,24 +1,36 @@
-/* eslint-disable react/jsx-no-constructed-context-values */
-/* eslint-disable react/prop-types */
-/* eslint-disable jsdoc/require-jsdoc */
 import React, {
-  createContext, useContext, useEffect, useState
+  createContext, ReactNode, useContext, useEffect, useMemo, useState
 } from 'react';
 import { GlobalTheme } from '@carbon/react';
 
-const ThemePreferenceContext = createContext();
+interface ThemeContextData {
+  theme: string,
+  setTheme: React.Dispatch<React.SetStateAction<string>>
+}
+
+const ThemePreferenceContext = createContext<ThemeContextData>({} as ThemeContextData);
 const currentMode = localStorage.getItem('mode');
 
+/** */
 function useThemePreference() {
   return useContext(ThemePreferenceContext);
 }
 
-function ThemePreference({ children }) {
+interface ThemePreferenceProps {
+  children?: ReactNode
+}
+
+/**
+ * @param root0
+ * @param root0.children
```

Comment:
You could use references to the parameters, using a notation like `children {ReactNode} nodes to be rendered.` Also don't forget to include a description to the function.